core/src/main/java/org/apache/iceberg/DeleteFileIndex.java

```diff
@@ -259,19 +258,16 @@ && allNonNull(dataNullCounts, field)) {
   }
 
   private static <T> boolean rangesOverlap(
-      Type.PrimitiveType type,
+      Types.NestedField field,
       ByteBuffer dataLowerBuf,
       ByteBuffer dataUpperBuf,
-      ByteBuffer deleteLowerBuf,
-      ByteBuffer deleteUpperBuf) {
-    Comparator<T> comparator = Comparators.forType(type);
+      T deleteLower,
+      T deleteUpper) {
+    Type.PrimitiveType type = field.type().asPrimitiveType();
+    Comparator<T> cmp = Comparators.forType(type);
     T dataLower = Conversions.fromByteBuffer(type, dataLowerBuf);
     T dataUpper = Conversions.fromByteBuffer(type, dataUpperBuf);
-    T deleteLower = Conversions.fromByteBuffer(type, deleteLowerBuf);
-    T deleteUpper = Conversions.fromByteBuffer(type, deleteUpperBuf);
-
-    return comparator.compare(deleteLower, dataUpper) <= 0
-        && comparator.compare(dataLower, deleteUpper) <= 0;
+    return cmp.compare(deleteLower, dataUpper) <= 0 && cmp.compare(dataLower, deleteUpper) <= 0;
```

Comment:
Here, maybe we can avoid the conversion for the upper. But this is a little improvement.

Reply:
Updated to match the checks we have for the file path.